### PR TITLE
pkg/console/subresource/route: Drop GetCanonicalHost and IsAdmitted

### PR DESCRIPTION
**Pull request by wking**
In favor of library-go's `IngressURI` helper.  The outgoing code results in error messages [like][1]:

    level=info msg=Cluster operator console Progressing is True with DefaultRouteSync_FailedAdmitDefaultRoute: DefaultRouteSyncProgressing: route "downloads" is not available at canonical host []

In that no-ingress case, the new code should produce error messages [like][2]:

    ... no ingress for host $HOST in route $NAME in namespace $NAMESPACE

which is more readable than the `[]` empty-ingress rendering.

IngressURI also avoids the need to assume the canonical ingress is `https://`.  But I haven't gone through and plumbed things to drop the `util.HTTPS` calls yet.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-network-operator/1156/pull-ci-openshift-cluster-network-operator-master-e2e-azure-ovn/1417036045987352576#1:build-log.txt%3A53
[2]: https://github.com/openshift/library-go/blob/7a65fdb398e28782ee1650959a5e0419121e97ae/pkg/route/routeapihelpers/routeapihelpers.go#L45

### CODE REVIEW

File: vendor/github.com/openshift/library-go/pkg/operator/staleconditions/remove_stale_conditions.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func NewRemoveStaleConditionsController(
 		conditions:     conditions,
 		operatorClient: operatorClient,
 	}
-	return factory.New().ResyncEvery(time.Second).WithSync(c.sync).WithInformers(operatorClient.Informer()).ToController("RemoveStaleConditionsController", eventRecorder.WithComponentSuffix("remove-stale-conditions"))
+	return factory.New().ResyncEvery(time.Minute).WithSync(c.sync).WithInformers(operatorClient.Informer()).ToController("RemoveStaleConditionsController", eventRecorder.WithComponentSuffix("remove-stale-conditions"))
```
Comment (wking) [inline vendor/github.com/openshift/library-go/pkg/operator/staleconditions/remove_stale_conditions.go:28]: I dunno why this is coming in with my bump, but it's from openshift/library-go#1036.  Maybe a previous commit rev'ed the pinned library-go commit, but forgot to update this vendored file?

Comment (wking) [reply vendor/github.com/openshift/library-go/pkg/operator/staleconditions/remove_stale_conditions.go:28]: Seems like it was #522:

```console
$ git show 24b7188b30319aabb1621062f25750b311bb5688 vendor/modules.txt | grep '#.*library-go'
-# github.com/openshift/library-go v0.0.0-20201123212217-43f358922ea0
+# github.com/openshift/library-go v0.0.0-20210330121117-68dd4a4c9d9e
```

**Review by spadgett — APPROVED** [review]
Thank you 👍 

/lgtm

**Comment by wking** [comment]
I dunno what that [`Interacting with the environment variable editor.When a variable is deleted: does not show any variables`][1] failure is about.  The words don't sound like "Route issue" to me, though.

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console-operator/568/pull-ci-openshift-console-operator-master-e2e-aws-console/1417208798119989248

**Pull request merged**